tests/phpunit/integration/Core/Modules/REST_Modules_ControllerTest.php

```diff
@@ -0,0 +1,521 @@
+<?php
+/**
+ * ModulesTest
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Modules;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Modules\Modules;
+use Google\Site_Kit\Core\REST_API\REST_Routes;
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Storage\User_Options;
+use Google\Site_Kit\Tests\TestCase;
+use Google\Site_Kit\Tests\FakeHttpClient;
+use WP_REST_Request;
+
+class REST_Modules_ControllerTest extends TestCase {
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->user         = $this->factory()->user->create_and_get( array( 'role' => 'administrator' ) );
+		$this->context      = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$this->options      = new Options( $this->context );
+		$this->user_options = new User_Options( $this->context, $this->user->ID );
+		$this->modules      = new Modules( $this->context, $this->options, $this->user_options );
+		wp_set_current_user( $this->user->ID );
+
+		// This ensures the REST server is initialized fresh for each test using it.
+		unset( $GLOBALS['wp_rest_server'] );
+		remove_all_filters( 'googlesitekit_rest_routes' );
+
+		$this->modules->register();
```

Comment:
We shouldn't need to register `Modules` here, but also this kind of hides the registration of the controller which should be used more explicitly here given this is its test case 👍 

I think we can remove this call and update the others below to register the controller (we'd want to store a reference to it here if we're not going to instantiate it in the tests as well).

Reply:
Sorry, I meant we should be registering the controller instead, not `Modules`.

Reply:
Just to confirm, this is updated and can be marked as resolved.

https://github.com/google/site-kit-wp/blob/43c436489a48caeead4c33809187b6823fb0aaaf/tests/phpunit/integration/Core/Modules/REST_Modules_ControllerTest.php#L41